internal_task_pollers.go

```diff
@@ -186,21 +174,20 @@ func (wtp *workflowTaskPoller) ProcessTask(task interface{}) error {
 		}, serviceOperationRetryPolicy, isServiceTransientError)
 
 	if err != nil {
+		wtp.metricsScope.Counter(metrics.DecisionResponseFailedCounter).Inc(1)
 		return err
 	}
-	return nil
 
+	wtp.metricsScope.Timer(metrics.DecisionResponseLatency).Record(time.Now().Sub(responseStartTime))
```

Comment:
I think it still make sense to include that time as part of decision response latency because any retry indicate there are issues with responding.